### PR TITLE
feat: Add beneficiary using Qr Code

### PR DESCRIPTION
**Pull request by dilpreet2028**
Loan Creation:

Please make sure these boxes are checked before submitting your pull request - thanks!
- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by dilpreet96** [comment]
@therajanmaurya please review

**Comment by dilpreet2028** [comment]
@therajanmaurya fixed navigation drawer according to the discussion

**Comment by therajanmaurya** [comment]
No, you don't need to change. I will review today. It looks good to me.

File: app/src/main/java/org/mifos/selfserviceapp/ui/activities/HomeActivity.java [hunk 1]
```diff
@@ -213,4 +219,23 @@ public void run() {
...
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
+                                           @NonNull int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        if (requestCode == CAMERA_PERMISSION && grantResults[0] == PackageManager.
+                PERMISSION_GRANTED) {
+            Log.d(HomeActivity.class.getName(), "permission granted");
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/activities/HomeActivity.java]: What if user will deny the permission?

File: app/src/main/java/org/mifos/selfserviceapp/ui/fragments/HomeFragment.java [hunk 2]
```diff
@@ -151,7 +151,25 @@ public void accountsClicked() {
 
     @OnClick(R.id.ll_transfer)
     public void transferClicked() {
-
+        String[] transferTypes = {getString(R.string.transfer), getString(R.string.
+                third_party_transfer)};
+        new MaterialDialog.Builder().init(getActivity())
+                .setTitle("Choose Transfer Type")
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/fragments/HomeFragment.java]: don't use hardcoded string, add in strings.xml

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/selfserviceapp/ui/activities/HomeActivity.java (hunk 1 above)
Comment (dilpreet2028) [reply app/src/main/java/org/mifos/selfserviceapp/ui/activities/HomeActivity.java]: @therajanmaurya changes made

File: app/src/main/java/org/mifos/selfserviceapp/ui/activities/HomeActivity.java [hunk 3]
```diff
@@ -217,4 +228,76 @@ public void run() {
...
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.CAMERA) !=
+                PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.CAMERA},
+                    CAMERA_PERMISSION);
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
+                                           @NonNull int[] grantResults) {
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/ui/activities/HomeActivity.java]: @dilpreet96  Please look this https://github.com/openMF/android-client/blob/master/mifosng-android/src/main/java/com/mifos/utils/CheckSelfPermissionAndRequest.java

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

**Comment by dilpreet2028** [comment]
@therajanmaurya changes made

**Review by therajanmaurya — APPROVED** [review]

**Pull request merged**